test_util/test_runner.py

```diff
@@ -120,37 +130,53 @@ def integration_test(
         add_env: a python dict with any number of key=value assignments to be passed to
             the test environment
     """
-    log.info('Transfering integration_test.py')
     test_script = pkg_filename('integration_test.py')
-    tunnel.remote_cmd(['mkdir', '-p', test_dir])
-    tunnel.write_to_remote(test_script, test_dir+'/integration_test.py')
+    pytest_docker = pkg_filename('docker/py.test/Dockerfile')
 
-    test_container_name = 'int_test_' + str(int(time.time()))
     dns_search = 'true' if test_dns_search else 'false'
-    test_cmd = [
-        'docker', 'run', '-v', test_dir+'/integration_test.py:/integration_test.py',
-        '--net=host', '--name='+test_container_name]
-    test_cmd.extend([
-        '-e', 'DCOS_DNS_ADDRESS=http://'+dcos_dns,
-        '-e', 'MASTER_HOSTS='+','.join(master_list),
-        '-e', 'PUBLIC_MASTER_HOSTS='+','.join(master_list),
-        '-e', 'SLAVE_HOSTS='+','.join(agent_list),
-        '-e', 'PUBLIC_SLAVE_HOSTS='+','.join(public_agent_list),
-        '-e', 'REGISTRY_HOST=127.0.0.1',
-        '-e', 'DCOS_VARIANT='+variant,
-        '-e', 'DNS_SEARCH='+dns_search,
-        '-e', 'AWS_ACCESS_KEY_ID='+aws_access_key_id,
-        '-e', 'AWS_SECRET_ACCESS_KEY='+aws_secret_access_key,
-        '-e', 'AWS_REGION='+region])
+    test_env = [
+        'DCOS_DNS_ADDRESS=http://'+dcos_dns,
+        'MASTER_HOSTS='+','.join(master_list),
+        'PUBLIC_MASTER_HOSTS='+','.join(master_list),
+        'SLAVE_HOSTS='+','.join(agent_list),
+        'PUBLIC_SLAVE_HOSTS='+','.join(public_agent_list),
+        'REGISTRY_HOST=127.0.0.1',
+        'DCOS_VARIANT='+variant,
+        'DNS_SEARCH='+dns_search,
+        'AWS_ACCESS_KEY_ID='+aws_access_key_id,
+        'AWS_SECRET_ACCESS_KEY='+aws_secret_access_key,
+        'AWS_REGION='+region]
     if add_env:
         for key, value in add_env.items():
-            extra_env = ['-e', key+'='+value]
-            test_cmd.extend(extra_env)
-    test_cmd.extend(['py.test', 'py.test', '-vv', ci_flags, '/integration_test.py'])
+            extra_env = key + '=' + value
+            test_env.append(extra_env)
+    test_cmd = 'py.test -vv ' + ci_flags + ' /integration_test.py'
+    cmd_script = gen.template.parse_str(BASH_TEMPLATE).render({'test_env': test_env,
+                                                               'test_cmd': test_cmd})
+    log.info('Building py.test image')
+    # Make a clean docker context
+    temp_dir = tempfile.mkdtemp()
+    with open(join(temp_dir, 'test_wrapper.sh'), 'w') as fh:
```

Comment:
```
from pkgpanda.util import write_string

write_string(join(temp_dir, 'test_wrapper.sh'), cmd_script)
```
I think that's a little cleaner (hides more API details which aren't the focus here), but fine either way.